libs/wingc/src/ast.rs

```diff
@@ -267,6 +267,11 @@ pub enum StmtKind {
 		iterable: Expr,
 		statements: Scope,
 	},
+	ForSequence {
+		iterator: Symbol,
+		sequence: Expr,
+		statements: Scope,
+	},
```

Comment:
This looks off to me, I don't think we should use different AST nodes to distinguish between `for x in <collection>` and `for x in <range>`. "sequence" in `ForSequence` should be the same as "iterable" in `ForLoop`.